SharedCode/src/commonMain/kotlin/ApplicationPresenter.kt

```diff
@@ -1,19 +1,56 @@
 package com.jetbrains.handson.mpp.mobile
 
-import kotlinx.coroutines.*
+import io.ktor.http.*
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.SupervisorJob
 import kotlin.coroutines.CoroutineContext
 
 class ApplicationPresenter: ApplicationContract.Presenter() {
 
     private val dispatchers = AppDispatchersImpl()
     private var view: ApplicationContract.View? = null
     private val job: Job = SupervisorJob()
+    private val stations: List<Station> = listOf(
+        Station("Kings Cross", "KGX"),
+        Station("York", "YRK"),
+        Station("Edinburgh Waverley", "EDB"),
+        Station("Leeds", "LDS"),
+        Station("Cattal", "CTL")
+    )
 
     override val coroutineContext: CoroutineContext
         get() = dispatchers.main + job
 
     override fun onViewTaken(view: ApplicationContract.View) {
         this.view = view
         view.setLabel(createApplicationScreenMessage())
+        view.setStations(stations);
+    }
+
+    override fun onTimesRequested() {
+        val departureStation = view?.getDepartureStation();
+        val arrivalStation = view?.getArrivalStation();
+
+        val now = "2020-10-14T19:30:00.000+01:00";
```

Comment:
I would question whether this is really 'now' 😛

Reply:
yeah it was supposed to be, but we struggled with getting the current DateTime. I meant to message you about that - I looked in gradle and we seem to have Klock already installed as a dependency but we can't use it within the code. Do you know how to fix that?

Reply:
Try adding `implementation 'com.soywiz.korlibs.klock:klock:1.10.0'` to the app build.gradle file

Reply:
Looking into it, seems to be behaving quite weirdly - will let you know when I've found a solution 🙂

Reply:
When you open the project in Android studio it probably comes up with a little popup in the lower right corner saying that it recommends updating Gradle, if you click on the 'update' link in that it comes up with a window telling you to update  Gradle to version 6.1.1, click 'Update'. It should then redo the Gradle build, and after that, it should start working properly. If that doesn't work, try adding `enableFeaturePreview('GRADLE_METADATA')` to the settings.gradle file as well, and rebuilding